### DISCUSSION TITLE
Where to find good Documentation how to compile IM

### DISCUSSION THREAD

**Discussion by arnold-b** [ImageMagick/ImageMagick#6867] [Development]
### ImageMagick version

7.X

### Operating system, version and so on

Ubuntu 22.04.3 LTS

### Description

Hi, is there anywhere a good Documentation which also show how to delegate and how to find out if it not working. 
I just try to get a simple compilation with jpeg and png and i dont succeed. 
All what i can find is documentation how to type ./configure and make. This is useless

Thanks

**Comment by urban-warrior** [comment]
The `./configure` script plays a crucial role in establishing the build environments for ImageMagick. It systematically assesses each potential delegate library to ascertain its compatibility with the intended ImageMagick build. The process involves identifying validated delegates and those that did not pass validation, with detailed information logged in a `config.log` file, which is readily accessible for inspection.

To illustrate, suppose the JPEG delegate library fails validation. In such cases, you can refer to the `config.log` file and specifically search for the entry related to JPEG. This log entry will elucidate the attempt to validate the JPEG library and provide insights into the reasons for its failure. Addressing the reported issues is paramount; once resolved, re-run the `configure` script. It is an iterative process, requiring successive attempts until all identified issues are rectified, resulting in successful validation of the library.

**Comment by fmw42** [comment]
See also https://imagemagick.org/script/advanced-linux-installation.php. Otherwise, search Google for any instructions for your flavor of Linux.

**Reply by arnold-b** [reply]
Thanks a lot. 
Jpeg ist not the problem. 
PNG i can not get running. i tried many different ways. Also installed libpng-1.6.31.tar.gz and installed it. But finaly libpng is never detected on IM
root@www:~/ImageMagick# ./configure | grep png
checking for libpng >= 1.0.0... no
checking for gs color device... png16m
checking for gs alpha device... pngalpha
  PNG               --with-png=yes                      no
  
  
  
root@www:~/ImageMagick/libpng-1.6.31# make install
make  install-am
make[1]: Verzeichnis „/root/ImageMagick/libpng-1.6.31“ wird betreten
make[2]: Verzeichnis „/root/ImageMagick/libpng-1.6.31“ wird betreten
 /usr/bin/mkdir -p '/usr/local/lib'
 /bin/bash ./libtool   --mode=install /usr/bin/install -c   libpng16.la '/usr/local/lib'
libtool: install: /usr/bin/install -c .libs/libpng16.so.16.31.0 /usr/local/lib/libpng16.so.16.31.0
libtool: install: (cd /usr/local/lib && { ln -s -f libpng16.so.16.31.0 libpng16.so.16 || { rm -f libpng16.so.16 && ln -s libpng16.so.16.31.0 libpng16.so.16; }; })
libtool: install: (cd /usr/local/lib && { ln -s -f libpng16.so.16.31.0 libpng16.so || { rm -f libpng16.so && ln -s libpng16.so.16.31.0 libpng16.so; }; })
libtool: install: /usr/bin/install -c .libs/libpng16.lai /usr/local/lib/libpng16.la
libtool: install: /usr/bin/install -c .libs/libpng16.a /usr/local/lib/libpng16.a
libtool: install: chmod 644 /usr/local/lib/libpng16.a
libtool: install: ranlib /usr/local/lib/libpng16.a
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/sbin" ldconfig -n /usr/local/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /usr/local/lib

**Comment by MarkSetchell** [comment]
It generally works better if you install the "pkgconfig/pkg-config" package first to help it find compiler/linker settings.
